### ISSUE TITLE
Coordinate Conditioning Implementation / Weights

### ISSUE THREAD

**Issue by hydrowan** [IBM/terramind#12]
Awesome model! I am happy to work on a TerraTorch PR to implement the coordinate conditioning in TerraMind, but would appreciate some guidance on my approach.

I reverse engineered what I believe to be the tokenizer setup based on comments on issues and the details within the paper.

What I've done:
- I've recreated the tokenizer as the paper hints. I had assumed `-90->+90`, lat range and `-180->180` lon, with a 0.25 degree grid. Each coord is added as a never-split token, extending the pretrained WordPiece / BERT tokenizer from 30,522 to 32,684.

```python
lat_tokens = [f"lat={v:+06.2f}"  for v in np.arange(-90.00,  90.0001, 0.25)]
lon_tokens = [f"lon={v:+07.2f}"  for v in np.arange(-180.00, 180.0001, 0.25)]
coord_tokens = lat_tokens + lon_tokens # Add to tokenizer

# add_special_tokens=True
```
- Aligned coordinates by quantizing with `q = round(lat_or_lon / 0.25) * 0.25`
- Patched the model code so every coord layer is init with that size, and set the normalization params to identity.
- However, encoder_embeddings.coords.token_emb  seems trimmed to 30,000 in the model weights, which suggests that my assumption is off or the coordinate token rows were not included in the released weights.

Is this the correct strategy? 
Was the coordinate embedding intentionally witheld from the released weights? If so what is the timescale for the release? Alternatively, when the dataset is released I can do a finetune to restore coordinate function with my current setup.

Thank you !

**Comment by blumenstiel** [comment]
Hi @hydrowan, cool that you are looking into the coords! I already started adding it to TerraTorch (pr https://github.com/IBM/terratorch/pull/676). However, the generation is not yet working correctly, so there is some error in the code. I guess you can already use it for fine-tuning, but the generation is not working yet. I hope that I find some time this week to finalize it.

**Comment by blumenstiel** [comment]
Hi @mojganmadadi, unfortunetly I did not find the time to finalize the pr yet, sorry if this leads to delays on your side. I hope to find some time in the next two weeks, but I cannot promise.

**Comment by blumenstiel** [comment]
Coordinates are supported now, see docs.
